### PR TITLE
Import active posts and votes to genesis #535,#536

### PR DESCRIPTION
**Pull request by zxcat**
+ reward pool, posts and votes import
+ prepare to set correct ram_payer #562
+ simplify insertion (remove unused and duplicated stuff)
+ simplify balances insertion

### CODE REVIEW

File: programs/create-genesis/genesis_create.cpp [hunk 1]
```diff
@@ -1009,6 +1021,104 @@ struct genesis_create::genesis_create_impl final {
...
+                beneficiaries.reserve(sz);
+                for (const auto& b: c.active.beneficiaries) {
+                    beneficiaries.emplace_back(beneficiary{get_name(b.account), b.weight});
+                }
+            }
+            pk = c.id;
+            db.insert(pk, get_name(c.author), mvo
+                ("id", pk)             // uint64
+                ("permlink", c.permlink.value(_plnk_map))        // string
+                ("ref_block_num", 0)   // uint64
```
Comment (s-medvedev) [inline programs/create-genesis/genesis_create.cpp]: `ref_block_num` was deleted in https://github.com/GolosChain/golos.contracts/pull/623/files#diff-f003f2f5988fe75d3350500d7f6dfcc6L183

File: programs/create-genesis/genesis_info.hpp [hunk 2]
```diff
@@ -42,9 +43,9 @@ struct genesis_info {
 
     struct parameters {
         stake_params stake;
+        posting_rules posting_rules;
```
Comment (s-medvedev) [inline programs/create-genesis/genesis_info.hpp]: Please, insert `posting_rules` in genesis-info.json (https://github.com/GolosChain/golos.contracts/blob/master/genesis/genesis-info.json.tmpl)

**Comment by zxcat** [comment]
needs to be updated to support new message structure, WIP

File: programs/create-genesis/genesis_info.hpp (hunk 2 above)
Comment (zxcat) [reply programs/create-genesis/genesis_info.hpp]: resolved in GolosChain/golos.contracts#630

File: programs/create-genesis/genesis_create.cpp (hunk 1 above)
Comment (zxcat) [reply programs/create-genesis/genesis_create.cpp]: fixed in 99f5b4c, also updated to move some fields to `permlink` table from current posting implementation

File: programs/create-genesis/genesis_create.cpp [hunk 3]
```diff
@@ -1003,6 +1018,120 @@ struct genesis_create::genesis_create_impl final {
...
+            db.insert(pk, authors[cid], mvo
+                ("id", pk)            // uint64_t
+                ("message_id", cid)    // uint64_t
+                ("voter", generate_name(_accs_map[_visitor.acc_id2idx[v.voter]]))         // name
+                ("weight", v.vote_percent)        // int16_t
+                ("time", time_point(v.last_update).time_since_epoch().count())          // uint64_t
+                ("count", v.num_changes)         // uint8_t
+                ("delegators", delegators)    // std::vector<delegate_voter>
+                ("curatorsw", w)     // base_t
+                ("rshares", to_fbase(v.rshares))       // base_t
```
Comment (kaynarov) [inline programs/create-genesis/genesis_create.cpp]: after [this](https://github.com/GolosChain/golos.contracts/pull/610/commits/551b85a6af2029bd4a64f8831465585ca9885ab9) commit, rshares is interpreted as a raw value

File: programs/create-genesis/genesis_create.cpp [hunk 4]
```diff
@@ -769,61 +792,60 @@ struct genesis_create::genesis_create_impl final {
...
-        insert_balance_record(gls_vest_account_name, gp.total_vesting_fund_steem, gls_pk, ram_payer);
-        insert_balance_record(gls_post_account_name, gp.total_reward_fund_steem, gls_pk, ram_payer);
-        // TODO: internal pool of posting contract
+        insert_balance_record(config::stake_account_name, _total_staked, sys_pk, name());
+        insert_balance_record(gls_vest_account_name, gp.total_vesting_fund_steem, gls_pk, name());
+        insert_balance_record(gls_post_account_name, gp.total_reward_fund_steem, gls_pk, name());
 
         // accounts GOLOS
-        asset total_gls = asset(0, golos_sym);
+        asset total_gls = asset(0, symbol(GLS));
```
Comment (s-medvedev) [inline programs/create-genesis/genesis_create.cpp]: In Golos dApp `GLS` symbol is missing now.

Comment (zxcat) [reply programs/create-genesis/genesis_create.cpp]: it's GOLOS:
https://github.com/GolosChain/cyberway/blob/3f9c2633b575711bc1f3acf365e6c8a9b47cb325/programs/create-genesis/genesis_create.cpp#L56

File: programs/create-genesis/genesis_create.cpp (hunk 3 above)
Comment (zxcat) [reply programs/create-genesis/genesis_create.cpp]: thx, fixed in 28683ce

**Pull request merged**